cdap-app-fabric/src/main/java/io/cdap/cdap/internal/credential/DefaultCredentialProviderContext.java

```diff
@@ -38,7 +39,13 @@ public class DefaultCredentialProviderContext implements CredentialProviderConte
   protected DefaultCredentialProviderContext(CConfiguration cConf, String providerName) {
     String prefix = String.format("%s%s.", CredentialProvider.SYSTEM_PROPERTY_PREFIX,
         providerName);
-    this.properties = Collections.unmodifiableMap(cConf.getPropsWithPrefix(prefix));
+    Map<String, String> props = cConf.getPropsWithPrefix(prefix);
```

Comment:
Can we just add it as a boolean to the DefaultCredentialProviderContext along with the corresponding method in the interface?

Reply:
ack, done.